packages/synapse-react-client/src/synapse-queries/doi/useDOI.ts

```diff
@@ -87,21 +87,23 @@ export function useCreateOrUpdateDOI(
       if (options?.onSuccess) {
         options.onSuccess(...args)
       }
-      const doi = args[0].doi!
-      queryClient.invalidateQueries({
-        queryKey: keyFactory.getDOIAssociationQueryKey(
-          doi.objectType!,
-          doi.objectId!,
-          doi.objectVersion,
-        ),
-      })
-      queryClient.invalidateQueries({
-        queryKey: keyFactory.getDOIQueryKey(
-          doi.objectType!,
-          doi.objectId!,
-          doi.objectVersion,
-        ),
-      })
+      const requestDoi = args[1].doi
```

Comment:
Use the request DOI to invalidate the React-Query cache--the response DOI was often null.